src/js/ratingPopup.js

```diff
@@ -0,0 +1,151 @@
+      Number(activeRating).toFixed(1);
+
+    for (let i = 0; i < activeRating; i++) {
+      stars[i].classList.add('rating-star--active');
+    }
+
+    for (let i = activeRating; i < 5; i++) {
+      stars[i].classList.remove('rating-star--active');
+    }
+  }
+}
+
+function onGiveRatingBtnClick() {
+  exerciseId = JSON.parse(
+    document.querySelector('button[data-exercise]').dataset.exercise
+  )._id;
+  refs.exercisePopUpBackdrop.classList.remove('is-open');
+  refs.exercisePopUpBackdrop.style.display = 'none';
+  refs.ratingPopUpBackdrop.classList.add('is-open');
+
+  refs.ratingPopUpContent.innerHTML = createRatingMarkup();
+
+  addRatingPopupListeners();
+}
+
+function closeRatingPopUp() {
+  refs.ratingPopUpBackdrop.classList.remove('is-open');
+  refs.exercisePopUpBackdrop.style.display = 'flex';
+  refs.exercisePopUpBackdrop.classList.add('is-open');
+
+  removeRatingPopupListeners();
+}
+
+function createRatingStarsMarkup() {
+  // <svg class="rating-star rating-star--active">
+  let starsMarkup = '';
+
+  for (let i = 0; i < 5; i++) {
+    starsMarkup += `
+    <div class="rating-star-container" data-id="${i + 1}">
+      <svg class="rating-star js-rating-star">
+        <use href="./images/icons.svg#icon-star"></use>
+      </svg>
+    </div>`;
+  }
+
+  return `
+    <div class="rating rating--dark">
+      <span class="rating-label js-rating-label">0.0</span>
+        ${starsMarkup}
+      </div>`;
+}
+
+function createRatingMarkup() {
+  const ratingStarsMarkup = createRatingStarsMarkup();
+
+  const markup = `
+    <p class="rating-data-parameter-name">Rating</p>
+    <div class="rating-rating-wrapper">
+    <div class="dark-container">
+      <div class="sub-container">
+      ${ratingStarsMarkup}
+      </div>
+    </div>
+    </div>
+  `;
+
+  return markup;
+}
+
+function addRatingPopupListeners() {
+  refs.ratingPopUpCloseBtn.addEventListener('click', closeRatingPopUp);
+  refs.ratingPopUpContent.addEventListener('click', onStarClick);
+  refs.ratingForm.addEventListener('submit', onRatingFormSubmit);
+}
+
+function removeRatingPopupListeners() {
+  refs.ratingPopUpCloseBtn.removeEventListener('click', closeRatingPopUp);
+  refs.ratingPopUpContent.removeEventListener('click', onStarClick);
+  refs.ratingForm.removeEventListener('submit', onRatingFormSubmit);
+}
+
+// Function to check if the button exists and add the event listener
+function checkForButton() {
+  const button = document.querySelector('.js-add-rating');
+  if (button) {
+    button.addEventListener('click', onGiveRatingBtnClick);
+    return true;
+  }
+  return false;
+}
+
+// MutationObserver to monitor DOM changes and check for the button
+const observer = new MutationObserver(mutations => {
+  mutations.forEach(mutation => {
+    if (mutation.type === 'childList' || mutation.type === 'subtree') {
+      checkForButton();
+    }
+  });
+});
+
+// Start observing the entire document
+observer.observe(document, {
```

Comment:
It's necesarry because buttons where replaces from static to dynamyc yestarday. I was fising that till 02:00 AM yesterday. If you lnow how to fix adding event listeners to the dynamix elements please suggest ot fix.